Riot/Modules/MatrixKit/Models/Room/MXKRoomDataSourceManager.m

```diff
@@ -202,9 +225,19 @@ - (void)roomDataSourceForRoom:(NSString *)roomId create:(BOOL)create onComplete:
 
     // If not available yet, create the room data source
     MXKRoomDataSource *roomDataSource = roomDataSources[roomId];
-
+    
+    // check if the room's dataSource has events with late decryption issues and destroys it
+    BOOL roomDataSourceToBeDestroyed = [roomDataSourcesToDestroy containsObject:roomId];
+    
+    if (roomDataSource && roomDataSourceToBeDestroyed && create) {
+        [roomDataSource destroy];
+        roomDataSources[roomId] = nil;
```

Comment:
I wonder if this should be reusing `closeRoomDataSourceWithRoomId` in case that ever gets tweaked and this is missed.

Reply:
So basically I found that the best spot to do it is before opening a room. Calling it on closing it won't cover a case when you receive the key once closed.
Other than that, I did not destroy everything when the user is inside the room.
Also I did found out the destroy is really necessary otherwise it won't work properly